### PR TITLE
feat(config): governed v8 defaults-flip frame — conservative preset, migration, rollback, inventory (#2504)

### LINKED ISSUE #2504: [Workstream F] PR 12 of 21: Flip only proven defaults in v8 with migration and rollback
**Roadmap:** F12 · **Workstream F: User workflows, host support and configuration**
**Sequence:** PR 12 of 21 (tracking slot; dependencies define execution order)
**Roadmap rebaseline:** 2026-09-05

The existing scope below remains required. Historical source/validation notes are context, not fresh execution evidence; the 2026-09-05 requirements refine the current closure contract.

**Validated against:** main @ 65be8637 (v7.162.1) on 2026-09-02 — re-anchor line citations before implementation if main has advanced.
**Source issues:** #1677 (closed into the 2026-09-02 roadmap re-baseline; see their closure comments).

## Outcome (observable behavior)
Every v8 default flip cites production evidence; migration and rollback are documented and tested.

## Required scope — independently testable obligations
- #1677: inventory every proposed default with production evidence citations; conservative preset; config migration; compatibility matrix; warnings; rollback; semver docs. Candidate inventory MUST explicitly include: auto_select_architect default posture after F1 ships (K3 UX-3), always-visible startup health banner (K3 UX-6), and free-tier model default resolution against the live catalog (K3 UX-7) — these three were previously unowned.
- Do not default-enable experimental resilience, autonomy, remote export, training capture, or sandbox behavior without their own exit evidence.

## Invariants and non-goals
- Evidence-gated flips only.

## Adversarial / edge cases
- Old config upgrade
- Explicit user override wins

## Required tests
- Upgrade paths
- Conservative preset
- Rollback

## Closure contract
- No unwired code, no silent deferral, no closure based only on tests that bypass registered production paths. The implementation PR closes this issue via "Closes #<this>", and closes a source issue only if its complete validated scope is satisfied.
- Ships a docs/releases/pending/<unique-slug>.md fragment; PR description includes the AGENTS.md invariant audit section.

## 2026-09-05 required additions and corrections

- Require #2585, the relevant supported-host/feature evidence from #2586, and the published #2490/#2491/#2503 quality decisions before changing defaults.
- Specify migration behavior and rollback for each default. Earlier PR slot number is not merge order; later-added prerequisites still apply.
- A limited single-host milestone must not be advertised as full Windows/macOS/Linux or Node/Bun qualification.

## Execution and closure contract

Treat PR n of total as a stable tracking slot, not merge order. The explicit dependencies govern execution; focused implementation issues supply evidence to later qualification issues without depending on those later issues closing. Re-anchor source and active-PR overlap before implementation. Close only the fully satisfied scope: wired production callers, meaningful focused validation and documented outcomes, with no silent deferral. For user-visible code changes, include the repository-required pending release fragment and invariant audit; no version/changelog hand edits. Verification-only residuals may close with concrete current evidence rather than an unnecessary code patch.

### PR DESCRIPTION
**Pull request by zaxbysauce**
Closes #2504

PR head: see the commit history (synced with main; release-please owns versions).

## Summary

Ships the governance frame required by issue #2504 (Workstream F PR 12 of 21, consolidating EPIC #1677) around the already-armed v8 defaults flips — the release-gated `auto_review.enabled` flip (approved burn-in pin + major>=8 gate) and the parallel-first new-plan default (v7.132.0). **No new default flip, no version changes.**

- **`preset` config key** (`"default" | "conservative"`, optional): the conservative preset restores the pre-flip (v7) defaults — `auto_review.enabled: false` and serial new plans — applied as the lowest-precedence layer in config resolution so explicit user values always win (including partial sections like `auto_review: { mode: "gate" }` and configs with an `agents` block, the installer's own shape).
- **Config doctor migration + warnings**: governed default changes surface as `defaults-flip` info findings (naming the kill switch and the preset) while `config_format_version < 3`, with a "Defaults changes (v8)" report section; `/swarm config doctor --fix` acknowledges them by stamping `config_format_version: 3` — idempotently; passive runs never write. This also fixes the gap where applied migrations re-advertised forever because the version was never written back.
- **Preset-aware new plans**: `save_plan` consults the resolved preset — under `conservative`, new plans default to serial; otherwise the v8 parallel-first default applies. Explicit profile keys always win; config-load failures fail open to the v8 default.
- **Inventory + trust-posture docs**: new `docs/defaults-governance.md` (per-flip production-evidence citations — #2585 live host proof, #2586 partial Windows-only cells honestly scoped, #2490/#2491/#2503 quality decisions, pinned cost baseline; K3 UX-3/UX-6/UX-7 dispositions; five non-goal families; kill switches; rollback; cost-delta statement); amended overgeneralized "all default false" claims in `architecture.md` / `installation.md` / `design-rationale.md`; Compatibility Matrix in `installation.md`; regenerated `opencode-swarm.schema.json` + `docs/configuration.md` table.
- **Review-hardening round** (swarm-pr-review on this PR): the conservative preset now also survives the user-config-alone recovery fallback (broken project config no longer defeats it), doctor summary counts include the new findings, the acknowledgment stamp strips a UTF-8 BOM and logs failures, and coverage was added for the markdown section, the stamp's user-config path, the fallback, and explicit-`default` semantics.

## Invariant audit

- 1 (plugin init): not touched — no `server()`-resolution path changes; the preset layer lives in the pre-existing synchronous `buildConfigWithMeta`; `bun run build` exit 0 + Node-ESM `await import('./dist/index.js')` OK; no new init-path I/O.
- 2 (runtime portability): touched — build + Node-ESM import verified green; no `bun:` imports introduced; plugin shape `{ id, server }` unchanged.
- 3 (subprocesses): not touched — no new spawns; the stamp reuses the file-local `atomicWriteFileSync` already used by `applySafeAutoFixes`.
- 4 (.swarm containment): not touched — the stamp writes the existing config file path under the existing backup flow; no new `.swarm/` artifacts.
- 5 (plan durability): not touched — `save_plan`'s change only selects the new-plan `parallelization_enabled` default; ledger/projection paths untouched; plan-schema default stays `false`.
- 6 (test_runner safety): not touched — validation used shell commands and file-scoped `bun test` only.
- 7 (test writing): touched — bun:test only; new files under the 500-line cap; `_internals` DI seam (no new `mock.module`); `canonicalMkdtemp` everywhere (`check:test-tmpdir` exit 0); `check:mock-cleanup` exit 0.
- 8 (session state): not touched — no new session-scoped state.
- 9 (guardrails/retry): not touched.
- 10 (chat/system msg): not touched — no chat-transform changes.
- 11 (tool registration): not touched — no new tools/commands/agents; `validateConfigKey` case added per the schema-introspection coverage test; `check:events` pass; `check:registry-citations` exit 0 (no new anchor drift).
- 12 (release/cache): touched per protocol — pending release fragment `docs/releases/pending/2504-v8-defaults-governance.md` shipped; no version/CHANGELOG/`.release-please-manifest.json` hand edits.

## Test plan

- Frozen-contract suites: `bun test tests/unit/config/conservative-preset.test.ts tests/unit/config/defaults-flip-upgrade.test.ts tests/unit/config/defaults-flip-rollback.test.ts tests/unit/commands/doctor-defaults-flip-markdown-2504.test.ts` -> 45 pass / 0 fail (preset schema/loader/release-seam contracts incl. agents-block + user-config fallback + explicit-`default` + fail-open unknown value; upgrade/stamp idempotency incl. user-config stamp path; per-flip kill switches; markdown section).
- Blast radius: `bun test src/services/config-doctor.test.ts tests/unit/services/config-doctor-migration-fix.test.ts tests/unit/config-doctor-startup-isolation.test.ts tests/unit/config/parallelization-default.test.ts tests/unit/config/loader-unknown-top-level-warn.test.ts tests/unit/config/adversarial-schema.test.ts tests/unit/scripts/generate-config-schema.test.ts tests/unit/agents/reviewer-auto-review-opt-in.test.ts tests/unit/review/cost-baseline.test.ts tests/unit/utils/atomic-write-ratchet.test.ts tests/unit/scripts/retention-registry-rows.test.ts tests/unit/hooks/delegation-gate-parallel-guidance-2532.test.ts tests/unit/config/auto-select-architect.test.ts` -> all pass (the single doctor-suite local failure is the documented pre-existing XDG-override artifact in `getConfigPaths`, clean on CI).
- Note on `src/services/config-doctor.test.ts` (changed file): the shared fixture now pins `config_format_version: 3` and the SC-006 absent-version test passes `undefined` explicitly, so the version-gating suite keeps testing the v1 path with explicit overrides.
- Gates: `bun run typecheck` -> exit 0; `bunx @biomejs/biome ci <changed files>` -> exit 0 (repo-wide 94 pre-existing findings identical at base — pristine-archive parity verified); `bun run check:events`, `check:mock-cleanup`, `check:invariants`, `check:test-file-cap`, `check:registry-citations`, `scripts/check-test-tmpdir.sh`, `scripts/check-test-clock.sh` -> all exit 0; `bun run build` -> exit 0; Node-ESM import of `./dist/index.js` -> OK; `scan-deferred.sh` -> clean; `bun run drift:check --enforce` -> one LOCAL-ONLY capture-hash finding (`pr-standards.yml` vs this machine's captured evidence; the workflow file is untouched by this diff and a pristine `git archive HEAD` run exits 0).
- Independent validation: 10-lane swarm review (6 base + 9 matched micro families), 2 reviewer shards, critic challenge on all HIGH findings; fix round re-run reviewer + final critic gates on the post-fix diff.

### CODE REVIEW

File: src/services/config-doctor.ts [hunk 1]
```diff
@@ -2167,6 +2234,31 @@ export function runConfigDoctor(
...
+					`Kill switch: set ${flip.killSwitch}. ` +
+					`Restore all v7 defaults with preset: "${flip.preset}". ` +
+					`Evidence and rollback: ${flip.evidence} (#2504).`,
+				severity: 'info',
+				path: flip.field,
+				currentValue: 'v7 default',
+				autoFixable: false,
+			});
+		}
+	}
```
[the remark this answers is not in the document]
Comment (zaxbysauce) [reply src/services/config-doctor.ts:2260]: Confirmed and fixed in ea1c3a86: the severity counts are now recomputed after the defaults-flip emission loop, so `summary.info` (and the `.swarm/config-doctor.json` artifact summary) include the new findings. Verified by probe: a v7-era config now reports Info: 2 matching the 2 defaults-flip rows, and `tests/unit/commands/doctor-defaults-flip-markdown-2504.test.ts` + the upgrade suite pin the behavior.

File: src/tools/save-plan.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ import * as fs from 'node:fs';
 import * as path from 'node:path';
 import type { ToolDefinition } from '@opencode-ai/plugin/tool';
 import { z } from 'zod';
+// QA gate check — first save-plan integration with profile store
+import { loadPluginConfigWithMeta } from '../config';
```
[the remark this answers is not in the document]
Comment (zaxbysauce) [reply src/tools/save-plan.ts]: Confirmed and fixed in ea1c3a86: the stray 'QA gate check' comment moved back above the `qa-gate-profile` import it describes; the config-loader import is documented by the `_internals` doc block.

**Comment by zaxbysauce** [comment]
## 🤖 Multi-Stage PR Review
**Pipeline:** MiniMax-M2.7-highspeed (orientation) (context pack) → MiniMax-M2.7-highspeed (explorer) + MiniMax-M2.7-highspeed (explorer B) (parallel explore, distinct lenses) → GLM-5-turbo (critique) ↔ GLM-5-turbo (critique) (cross-critique) → MiniMax-M2.7-highspeed (fallback arbiter) (arbiter: blind-spot + synthesize)
**Commit reviewed:** `ea1c3a862a37`

---

Looking at this PR carefully — it's the governance frame for v8 defaults flips (#2504): a `preset` config key, config-doctor migration/acknowledgment, preset-aware `save_plan`, docs, and tests. Let me challenge every confirmed finding with evidence.

---

### 🔍 PR Intent

Reconstructed from issue #2504, PR description, and commit messages:

- **O-001**: Add `preset` top-level config key (`"default" | "conservative"`) — conservative restores pre-flip v7 defaults as lowest-precedence layer
- **O-002**: Config doctor surfaces `defaults-flip` info findings for unacknowledged configs (`config_format_version < 3`); `--fix` stamps `config_format_version: 3` idempotently
- **O-003**: `save_plan` consults resolved preset for new-plan `parallelization_enabled` default
- **O-004**: `docs/defaults-governance.md` inventories every governed flip with evidence, kill switches, rollback
- **O-005**: Amend stale "all default false" overgeneralizations in architecture.md / installation.md / design-rationale.md
- **O-006**: Regenerate schema and generated docs

---

### 📦 Implementation Summary

The PR adds `preset: "default" | "conservative"` to the schema, applies `CONSERVATIVE_PRESET_BASE` (`auto_review.enabled: false`) as a deep-merge base layer in `buildConfigWithMeta` step 3b (and mirrors it in the user-config-alone fallback path), adds `DEFAULT_FLIPS` table to `config-doctor.ts` for info-severity findings, adds a stamp in `runConfigDoctorWithFixes` to acknowledge, adds preset-aware logic in `save_plan`, updates three doc files to describe the actual default posture, and ships three new test files plus test additions.

---

### ✅ / ⚠️ / ❌ Intended vs Actual

| Obligation | Status | Evidence |
|---|---|---|
| O-001 | `SUPPORTED` | `src/config/schema.ts:3667` — `preset` enum added; `loader.ts:707` — base-layer injection |
| O-002 | `SUPPORTED` | `config-doctor.ts:2234` — `DEFAULT_FLIPS` loop; `config-doctor.ts:2762` — stamp logic |
| O-003 | `SUPPORTED` | `save-plan.ts:942` — `conservativePresetActive` flag; `save-plan.ts:1011` — default injected |
| O-004 | `SUPPORTED` | `docs/defaults-governance.md` — full inventory with evidence and kill switches |
| O-005 | `SUPPORTED` | `architecture.md:1925`, `installation.md:809`, `design-rationale.md:303` — all amended |
| O-006 | `SUPPORTED` | `opencode-swarm.schema.json` regenerated; `docs/configuration.md` table updated |

---

### 🚨 Confirmed Findings

> Only findings with concrete PR-introduced defects. Everything below is grounded in exact file:line evidence.

#### [MEDIUM] `xdgDir` temp directory leaks when test throws before cleanup

- **Location**: `tests/unit/config/defaults-flip-upgrade.test.ts:35` (and line 161)
- **Why it matters**: `xdgDir` created via `canonicalMkdtemp` in `beforeEach` has no `afterEach` cleanup. If a test body throws before its own `fs.rmSync` call (line 161's `dir`), `xdgDir` leaks to disk permanently for the test process lifetime. The same pattern exists for the `dir` created inside the `--fix stamps USER config` test at line 161 — `fs.rmSync(dir)` is inline, not in `afterEach`.
- **Evidence**: `beforeEach` (line 35) creates `xdgDir` but `afterEach` (line 35–43) only restores `XDG_CONFIG_HOME` and cleans `xdgDir` itself — it does **not** clean the `dir` variables created inside individual tests at lines 58–65 and 161. The parallelization-default test (line 20–53) has the same pattern fixed with an explicit `fs.rmSync(xdgDir, ...)` in `afterEach`; this file does not.
- **Fix direction**: Add `fs.rmSync(xdgDir, { recursive: true, force: true })` to `afterEach` (mirroring `parallelization-default.test.ts:50–54`), and move the `--fix stamps USER config` test's `fs.rmSync(dir, ...)` inline call into `afterEach`.

#### [MEDIUM] `stripUnrecognizedKeys` receives wrong input in user-config fallback path

- **Location**: `src/config/loader.ts:845`
- **Why it matters**: In the user-config-alone fallback path, `stripUnrecognizedKeys` now receives `userPresetReady` (post-migration + post-conservative-base) instead of `userSanitized`. The conservative base layer injects `auto_review: { enabled: false }` *before* `safeParse` is called. If the user's partial `auto_review` section (e.g. `{ mode: "gate" }`) then triggers a partial-success parse with unexpected Zod issues, `stripUnrecognizedKeys` will strip keys based on `userPresetReady`'s error shape — which includes issues from the pre-filled section — potentially stripping different keys than it would have from `userSanitized`.
- **Evidence**: `loader.ts:815–822` — `userPresetReady` is constructed from `userPresetMigrated.preset === 'conservative' ? deepMergeFn(CONSERVATIVE_PRESET_BASE, userPresetMigrated) : userPresetMigrated`, then passed to `safeParse` at line 824. The resulting `userParseResult.error` is the error from parsing `userPresetReady`, whose shape includes issues from the pre-filled base. Line 845 then passes `userPresetReady` (not `userSanitized`) to `stripUnrecognizedKeys`.
- **Fix direction**: Use `userPresetMigrated` (pre-base-layer) as the strip input, matching what would have happened on the normal merge path where the fallback ladder calls `stripUnrecognizedKeys(userSanitized, ...)`. The conservative base is already applied to the **parsed result** via the deep-merge before Zod — it does not need to be in the strip input.

#### [LOW] `Infinity` bypasses stamp idempotency guard silently

- **Location**: `src/services/config-doctor.ts:2795`
- **Why it matters**: `shouldStamp` is `typeof currentVersion !== 'number' || !Number.isInteger(currentVersion) || currentVersion < 0 || currentVersion < CURRENT_CONFIG_FORMAT_VERSION`. `Infinity < 3` is `false`, so `shouldStamp = false` — the stamp is silently skipped, leaving `Infinity` as the config version. A future bump of `CURRENT_CONFIG_FORMAT_VERSION` past `Infinity` (impossible today but not structurally prevented) would still not restamp. The schema caps at `9007199254740991` but the guard doesn't explicitly reject `Infinity` — `Infinity` passes `typeof === 'number'` and `Number.isInteger(Infinity)` is `false`.
- **Evidence**: `config-doctor.ts:2795` — the guard chain; `Infinity` triggers `!Number.isInteger(Infinity)` → `false`, then `Infinity < 0` → `false`, then `Infinity < 3` → `false`, so the overall expression is `false`.
- **Fix direction**: Add an explicit `!Number.isFinite(currentVersion)` guard (or use `Number.isSafeInteger`) to reject `Infinity` and `-Infinity` explicitly.

---

### 🔬 Unverified but Plausible Risks

> These look suspicious but cannot be structurally proven without runtime validation.

- **Risk**: `deepMergeFn` mutates `CONSERVATIVE_PRESET_BASE` in place.
  - *Why suspicious*: `src/config/loader.ts:711` — `deepMergeFn(CONSERVATIVE_PRESET_BASE, mergedRaw)` is called with `CONSERVATIVE_PRESET_BASE` as first argument. If `deepMergeFn` mutates its first argument, the constant is corrupted for the process lifetime.
  - *What would verify it*: Inspect `deepMergeFn` body for in-place mutation, or add `Object.freeze` + runtime `deepEqual` assertion in a stress test with multiple loader invocations.

- **Risk**: The conservative preset fallback in step 3b could have XDG/read-order issues.
  - *Why suspicious*: The fallback in `loader.ts:815` is a separate code path (fires when project config is unparseable). The user-config fallback is a less-tested execution path than the normal merge ladder.
  - *What would verify it*: Integration test that writes a broken project config + a conservative user config and asserts the user fallback correctly materializes `auto_review.enabled: false`.

---

### 🧪 Test / Coverage Gaps

- **Gap**: `defaults-flip-rollback.test.ts` tests `resolveAutoReviewConfig` at a simulated v8 release but does **not** test the full loader path (`loadPluginConfig`) with the conservative preset — only the direct unit function. A regression in `buildConfigWithMeta`'s step 3b deep-merge would not be caught.
  - *Evidence*: `defaults-flip-rollback.test.ts:24` — direct `resolveAutoReviewConfig` call; no `loadPluginConfig` or `loadPluginConfigWithMeta` in the file.
- **Gap**: The conservative preset rollback test does not verify that a v8-era config **without** a preset key still resolves to `enabled: true` at v8. Without this negative assertion, a future regression that makes the preset become the default (instead of the absent/omitted case) would go undetected.
  - *Evidence*: `defaults-flip-rollback.test.ts` — all three tests either set an explicit `enabled` or set `preset: 'conservative'`; no test for `{}` (no preset, no explicit key) at v8.

---

### 📋 Shipped-vs-Claimed Gaps

_None found._ The `docs/defaults-governance.md` inventory covers both flip entries with evidence citations; the kill switches are documented and tested; the conservative preset behavior is described accurately in all three amended docs files; the Compatibility Matrix correctly shows all four config scenarios.

---

### 📝 Merge Recommendation

**[APPROVE_WITH_FIXES]**

Two medium findings (temp-dir leak, wrong strip input) and one low (Infinity bypass) are real, fixable defects introduced by this PR. No CRITICAL findings survive the challenge.

| Check | Result |
|---|---|
| No CRITICAL findings | ✅ |
| No unresolved STEALTH_CHANGE | ✅ |
| No UNSUPPORTED obligations | ✅ |
| Test coverage adequate | ⚠️ 2 coverage gaps (not blockers) |
| No hardcoded secrets | ✅ |
| All async errors handled | ✅ |
| Input validation present | ✅ |
| No broken agent role boundaries | ✅ |
| Prompt format contracts intact | ✅ |
| Lockfile consistent | ✅ |

---

### 🔁 Validation provenance

**Findings KILLED (refuted with evidence)**:
- `loader.ts:199` null guard — `raw` is a required `Record<string, unknown>` from callers; contract guarantees non-null
- `loader.ts:822` type safety — Zod `safeParse` failure always produces defined `error`; TypeScript's type narrowing is correct
- `loader.ts:845` input diverge — intentional; fallback must mirror steps 3/3b for conservative preset survival per PRR-004 review contract
- `schema.ts:622` plain string type — enum values validated by `safeParse` before reaching `autoReviewEnabledByRelease`; any invalid string is rejected earlier
- `schema.ts:639` null guard — `context` has default `{}`; `preset` is typed optional/string from Zod
- `doctor.ts:186` null guard — `findings` is a required `ConfigFinding[]` field on `ConfigDoctorResult`
- `config-doctor.test.ts:2926` test correctness — both `{}` and `{ config_format_version: undefined }` produce absent key in parsed object; equivalent test path
- `config-doctor.ts:105` data mismatch — `execution_profile.parallelization_enabled` is the correct plan-file path; the kill switch description says "(per plan)" which is accurate
- `save-plan.ts:954` error handling — intentional fail-open design; documented in PR description and code comment
- `defaults-flip-upgrade.test.ts:78` stale data — file read + parse is synchronous; no async flush gap exists
- `defaults-flip-upgrade.test.ts:102` malformed version — schema rejects non-integer versions (`minimum: 0, maximum: 9007199254740991`)
- `parallelization-default.test.ts:53` silent catch — intentional; ENOENT is expected; Bun test runner isolates each file
- `parallelization-default.test.ts:43` state leakage — `afterEach` restores `XDG_CONFIG_HOME` in a try/catch before deletion; safe
- `doctor.ts:189` markdown injection — `DEFAULT_FLIPS` descriptions are static strings from internal constants; not user-derived
- `loader.ts:711` deep-merge mutation — CONSERVATIVE_PRESET_BASE is Readonly and documented as a contract; the code is written assuming pure semantics
- `loader.ts:825` inconsistent input — `userPresetReady` is the correct input because the conservative base must be in the error-provenance path
- `schema.ts:672` runtime mutability — documented as a design constraint; not a runtime bug
- `index.ts:1680` kill-switch bypass — `autoReviewEnabledByRelease` checks `context.preset === 'conservative'` first; release gate never runs under conservative
- `save-plan.ts:944` logic null guard — empty string is intentional fallback to CWD; documented behavior
- `config-doctor.ts:2796` silent noop — no file to stamp exists; no-op is correct
- `config-doctor.ts:2817` fail-open stamp — intentionally non-fatal; documented
- `conservative-preset.test.ts:269` isolation gap — catch in afterEach still runs the restore; file-scoped runner
- `conservative-preset.test.ts:261` assertion weakness — intentional DI seam test; the seam itself is tested separately
- `defaults-flip-rollback.test.ts:25` XDG isolation — confirmed existing gap; not introduced by this PR (the PR's own test file `conservative-preset.test.ts` correctly adds XDG isolation)
- `defaults-flip-upgrade.test.ts:35` resource leak — KILLED as null, REINTRODUCED above as confirmed MEDIUM
- `defaults-flip-upgrade.test.ts:161` resource leak — KILLED as null, REINTRODUCED above as confirmed MEDIUM
- `parallelization-default.test.ts:20` concurrency — KILLED; Bun runs tests in the same file serially by default

**Findings SURVIVING (with new evidence)**:
- `defaults-flip-upgrade.test.ts` temp-dir leaks → **MEDIUM** (confirmed above)
- `loader.ts:845` wrong strip input → **MEDIUM** (confirmed above)
- `config-doctor.ts:2795` Infinity bypass → **LOW** (confirmed above)

**Blind-spot findings added**: None — the PR touches a well-tested surface with good review coverage; no new structural defects found beyond the confirmed items above.

**Documentation findings (docs only, not PR-introduced code defects)**: All five documentation items (compatibility matrix rows, migration steps inconsistency, v7 release qualification, behavioral documentation changes) describe the PR's intentional documented behavior. None are code defects.

---
<sub>🔒 Reviewed by a 3-model cross-family adversarial debate (architect → dual-lens parallel explorers → cross-critique → arbiter) for high recall with low false-positive noise. Findings are advisory — verify before acting.</sub>
